doc/s3ip_sysfs/s3ip_sysfs_specification.md

```diff
@@ -0,0 +1,239 @@
+/sys_switch/temp_sensor/temp[n]/min| R/W| int| Alarm recovery threshold, unit: millidegree Celsius
+/sys_switch/temp_sensor/temp[n]/value| RO| int| Current temperature, unit: millidegree Celsius
+
+### 3. Voltage sensor sysfs
+The sysfs path of the voltage sensor information must be /sys_switch/vol_sensor/
+
+*Table3-1 Voltage Sensor SYSFS property *
+
+|Sysfs path|Permissions|Data type|Description|
+|-|-|-|-|
+/sys_switch/vol_sensor/number|RO|int| Total number of voltage sensors
+/sys_switch/vol_sensor/vol[n]/alias|RO|string| Identifying the location of the voltage sensor
+/sys_switch/vol_sensor/vol[n]/type|RO|string| Model of the voltage sensor
+/sys_switch/vol_sensor/vol[n]/max|R/W|int| Alarm threshold, unit: mV
+/sys_switch/vol_sensor/vol[n]/min|R/W|int| Alarm recovery threshold, unit: mV
+/sys_switch/vol_sensor/vol[n]/range|RO|int| Voltage output error value, such as +-500mV
+/sys_switch/vol_sensor/vol[n]/nominal_value|RO|int| Nominal value of supply voltage, unit: mV
+/sys_switch/vol_sensor/vol[n]/value|RO|int| Voltage value, unit: mV
+
+### 4. Current sensor sysfs
+The sysfs path of the current sensor information must be /sys_switch/curr_sensor/
+
+*Table4-1 Current Sensor SYSFS property*
+
+|Sysfs path|Permissions|Data type|Description|
+|-|-|-|-|
+/sys_switch/curr_sensor/number|RO|int| Total number of current sensors
+/sys_switch/curr_sensor/curr[n]/alias|RO|string| Identifying the location of the current sensor
+/sys_switch/curr_sensor/curr[n]/type|RO|string| Current sensor model
+/sys_switch/curr_sensor/curr[n]/max|R/W|int| Alarm threshold, unit: mA
+/sys_switch/curr_sensor/curr[n]/min|R/W|int| Alarm recovery threshold, unit: mA
+/sys_switch/curr_sensor/curr[n]/value|RO|int| Current value, unit: mA
+
+### 5. Syseeprom information sysfs
+The sysfs path of the syseeprom information must be /sys_switch/syseeprom
+
+*Table5-1 Syseeprom info SYSFS property*
+
+|Sysfs path|Permissions|Data type|Description|
+|-|-|-|-|
+/sys_switch/syseeprom|RO|int|The file content conforms to the ONIE standard binary
+
+### 6. Fan information sysfs
+The sysfs path of the fan information must be /sys_switch/fan/
+
+*Table6-1 fan info sysfs property*
+
+|Sysfs path|Permissions|Data type|Description|
+|-|-|-|-|
+|/sys_switch/fan/number |RO| int| Total number of fans
+|/sys_switch/fan/fan[n]/model_name |RO| string| Fan name
+|/sys_switch/fan/fan[n]/serial_number |RO| string| Fan serial number
+|/sys_switch/fan/fan[n]/part_number |RO| string| Fan Part Number
+|/sys_switch/fan/fan[n]/hardware_version |RO| string| Fan hardware version number
+|/sys_switch/fan/fan[n]/motor_number |RO| int| Number of fan motors
+|/sys_switch/fan/fan[n]/direction  |RO| enum| The duct types are defined as follows:<br>0: F2B, forward air duct <br>1: B2F, rear duct
+|/sys_switch/fan/fan[n]/ratio  |R/W| int| Motor speed percentage, value range 0-100
+|/sys_switch/fan/fan[n]/motor[n]/speed |RO| int| Speed value,unit: RPM

```

Comment:
@tianshangfei 
Are you considering a Fantray as one fan unit? Or Fan[n] refers to individual fans inside each Fantray?

I have some confusion about the "Number of Fan Motors" field. If a Fantray is one fan unit, then "Front" and "Rear" fans can be marked as two motors which have different speed, max & min speeds, spped tolerance etc.

Reply:
The definitions of fans and motors are as follows: a device contains n fan units, and a fan unit contains 1-2 motors. Each motor may have a fault, so the speed, maximum and minimum speed, speed tolerance, etc. are collected by the motor. "Front" and "back" are used to identify the direction of the fan unit. F2B is the abbreviation for front-to-back airflow.